src/pages/index.js

```diff
@@ -65,25 +132,81 @@ function setFormInfo(nameSelector, detailsSelector) {
 }
 
 function createCard(data) {
-  const cardElement = new Card(data, handleImageClick , "#cards-template");
+  const cardElement = new Card(data, handleImageClick , "#cards-template" , handleDeleteClick, handleLikeClick);
   return cardElement.generateCard();
 }
 
+function handleLikeClick(card) {
+  if(card.isLiked) {
+    api.unlikeCard(card._id).then((res) => {
+      // update the UI
+
+      card.updateIsLiked(res.isLiked);
+    })
+  } else {
+    api.likeCard(card._id).then((res) => {
+      // update the UI
+      card.updateIsLiked(res.isLiked);
+    })
+  }
+}
+
 function handleImageClick(imgData) {
   previewModal.open(imgData);
 }
 
+function handleDeleteClick(card) {
+  console.log(card);
+  deleteCardModal.open();
+  deleteCardModal.setHandleConfirm(() => {
+    return api
+      .deleteCard(card._id)
+      .then(() => {
+        card._handleDeleteButton();
+        deleteCardModal.close();
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  });
+}
+
 function handleProfileFormSubmit(inputValues) {
-  updateUserInfo(inputValues);
+  profileEditForm.renderLoading(true)
+  api.updateUserInfo(inputValues)
+  .then((res) => {
+ // send request to PATCH /users/me
+  // only update the page after succesful request
+//  updateUserInfo(res);
+  currentUserInfo.setUserInfo(res)
   profileEditForm.close();
   formValidators["profile-edit-form"].resetValidation();
+  })
+  .catch((err) => {
+    console.log(err)
+  })
+  .finally(() => {
+    profileEditForm.renderLoading(false);
+  })
+ 
+  
 }
 
 function handleAddCardFormSubmit(inputValues) {
-  const cardElement = createCard(inputValues);
+  addCard.renderLoading(true)
+api.addNewCard(inputValues)
```

Comment:
You need to format the markup to show **indentation** and to make the code more readable. You can do that with `prettier` or a combination `SHIFT + ALT + F`